app/src/test/java/org/simple/clinic/setup/SetupActivityEffectHandlerTest.kt

```diff
@@ -42,12 +45,13 @@ class SetupActivityEffectHandlerTest {
     // given
     whenever(onboardingCompletePreference.get()) doReturn true
     whenever(userDao.userImmediate()).doReturn<User?>(null)
+    whenever(appConfigRepository.currentCountry()) doReturn None
 
     // when
     testCase.dispatch(FetchUserDetails)
 
     // then
-    testCase.assertOutgoingEvents(UserDetailsFetched(hasUserCompletedOnboarding = true, loggedInUser = None))
+    testCase.assertOutgoingEvents(UserDetailsFetched(hasUserCompletedOnboarding = true, loggedInUser = None, userSelectedCountry = None))
```

Comment:
Constructors in data classes expose too much information. Hence we should always consider exposing them via factory instance creation functions. These set of parameters have a meaning, but as a maintainer, I have to try hard to understand what it means. Also, when we keep adding new parameters to these functions, the tests and production code can get out of sync pretty quickly.

Reply:
For this specific event, all three properties will always have to be set together since this is an event. The tests have been updated to reflect this.